proposals/solid-oidc/index.bs

```diff
@@ -181,9 +181,16 @@ SHOULD use a WebID as their Client ID.
 
 ## WebID ## {#clientids-webid}
 
-The WebID itself MUST resolve to an RDF document. That document SHOULD be available as JSON-LD with an `@context`
-that includes `https://www.w3.org/ns/solid/oidc-context.jsonld` but which otherwise produces a JSON serialization of an OIDC
-client registration, per the definition of client registration metadata from [[!RFC7591]] section 2.
+A client's WebID resource MUST be serialized as an `application/ld+json` document when
+the request includes an Accept header, unless content negotiation requires a different
+outcome.
+
+The serialized JSON form of a client WebID SHOULD use the normative JSON-LD `@context`
+provided at `https://www.w3.org/ns/solid/oidc-context.jsonld` such that the resulting
+document produces a JSON serialization of an OIDC client registration, per the
+definition of client registration metadata from [[!RFC7591]] section 2.
+Client WebID documents MAY augment the normative context with additional `@context`
+definitions but MUST NOT override or change the terms defined by the normative context.
```

Comment:
This language is identical to Activity Streams 2.0 https://www.w3.org/TR/activitystreams-core/#jsonld
> Implementations MAY augment the provided @context with additional @context definitions but MUST NOT override or change the normative context.

Reply:
And?

Reply:
I think it it makes it clear that one can extend that context as long as they don't overwrite anything from the specified one. Could you propose alternative way of expressing that and explain why would you consider it an improvement?

Reply:
Still doesn't answer my questions.

Is there anything (by default) preventing additional context to be used? What happens if the spec doesn't say it but implementations use additional context? What happens if an implementation completely overrides the context? Is "MAY" proposed because it enhances something given implementation experience? Is that what your implementation is doing?

AS2 makes the statements because JSON is the required serialization and that it should be consistent with compacted JSON-LD using at least the AS2 context. Telling JSON publishers to take care not to override is reasonable guidance. On the other hand, Solid OIDC spec (above) uses a MUST for JSON-LD and as long as the required context is provided, nothing restricts additional context use. If there is any overriding going, it breaks the initial requirement. The spec doesn't need to say how not to conform.

Reply:
>  If there is any overriding going, it breaks the initial requirement.

I don't think we should rely on implementer arriving at this conclusion. One could get impression of conforming by only including mandated context. Just to include another one that overwrites parts of it without giving it another thought.

I think this helps guiding implementation. I can't see any disadvantage in including it in the spec.

Reply:
The minimum assumption about implementers that are going beyond Solid OIDC's interop requirements is that they're doing something because they 1) have a need, and 2) have read the JSON-LD spec on how to using multiple contexts or understand potential errors. Providing guidance about ways that an implementation may by choice be non-conforming does not require normative text here.

Reply:
This `MAY` / `MUST NOT` section has been entirely removed. Here we are relying on the JSON-LD 1.1 feature of `"@protected": true` to convey the same thing without having to state it in normative language.